src/net.cpp

```diff
@@ -845,6 +852,14 @@ static bool CompareNodeTXTime(const NodeEvictionCandidate &a, const NodeEviction
     return a.nTimeConnected > b.nTimeConnected;
 }
 
+// Pick out the potential block-relay only peers, and sort them by last block time.
+static bool CompareNodeBlockRelayOnlyTime(const NodeEvictionCandidate &a, const NodeEvictionCandidate &b)
```

Comment:
If this PR was built on top of #19316, you could test directly for block-relay-only peers. I think that'd be a cleaner implementation.

Reply:
We don’t currently have a way in our codebase to know that an inbound peer is blocks-only; that is a concept we can only know for outbound peers. We can infer it for inbound peers, as this PR does, but the right way to do this in the future would be to negotiate it at startup, so both sides know that a connection will stay blocks only.

I believe that PR is only cleaning things up for outbound blocks-only peers?

Reply:
Yes, you're right. 19316 only changes things for outbound peers.